### PR TITLE
New workflow to make benchmarks to run on PR commits

### PR DESCRIPTION
**Pull request by parthb83**
## Description
Added a new file `pr_benchmarks.yml` to run benchmarks on pull request commits and then comment on the pull request with the results.

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [ ] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: .github/workflows/pr_benchmark_comment.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
```
Comment (phschiele) [inline .github/workflows/pr_benchmark_comment.yml:20]: @parthb83 Didn't we want to use the marketplace action using this ID instead of this handmade script?

Comment (parthb83) [reply .github/workflows/pr_benchmark_comment.yml:20]: Yes but the workflow failed as it couldn't define "context" in `context.workflow_run.payload.id`. Therefore, I used this script that was given in `workflow_run` documentation.

Comment (phschiele) [reply .github/workflows/pr_benchmark_comment.yml:23]: Resolved through [b96aef4](https://github.com/cvxpy/cvxpy/pull/1810/commits/b96aef47f9aaefe3bac1cbdf5c23ccf28281ea53)

**Review by phschiele — APPROVED** [review]
LGTM! 
@cvxpy/maintainers 
TL;DR:
1. install asv + cvxpy and run `asv continuous` to compare PR HEAD vs. master.
2. run `asv compare` to create a comparison table.
3. as this is running on a read-only view of the cvxpy repo, the comment is created by another workflow that has write access to the comments.

I suggest merging this PR and creating a dummy PR to see it in action.

This is how the table would look like for the last commit in this PR:
[9bd30a6](https://github.com/cvxpy/cvxpy/commit/9bd30a6f379ee01fedac11c3f982cd97ddadea05) is the current master commit, and ff81075b is a "simulated" merge commit of the PR into the current master i.e., merging https://github.com/cvxpy/cvxpy/commit/e93e18c6ca4bbacdbffba28a091f37a21de67759 into https://github.com/cvxpy/cvxpy/commit/9bd30a6f379ee01fedac11c3f982cd97ddadea05. This is relevant if the PR was created based on an outdated version of master.
image

When the list of benchmarks grows larger, we can only show the benchmarks that do exhibit significant changes.

**Review by rileyjmurray — APPROVED** [review]

**Comment by rileyjmurray** [comment]
Since this doesn't change the public API we don't need two maintainers to approve. So I'll go ahead and merge. (The failing tests are unrelated.)
